pkg/issuer/acme/dns/util/wait.go

```diff
@@ -66,32 +70,42 @@ func getNameservers(path string, defaults []string) []string {
 	return systemNameservers
 }
 
-// Update FQDN with CNAME if any
-func updateDomainWithCName(r *dns.Msg, fqdn string) string {
-	for _, rr := range r.Answer {
-		if cn, ok := rr.(*dns.CNAME); ok {
-			if cn.Hdr.Name == fqdn {
-				logf.V(logf.DebugLevel).Infof("Updating FQDN: %s with its CNAME: %s", fqdn, cn.Target)
-				fqdn = cn.Target
-				break
+// Update FQDN with CNAME if any, it will follow CNAME records till it hits a non-CNAME.
+// this will error if there is a recursive CNAME in the chain
+func updateDomainWithCName(fqdn string, nameservers []string, fqdnChain ...string) (string, error) {
```

Comment:
I'm confused by the function name `updateDomainWithCName`, it kind of suggests we are "updating" something on the name server 🤔

Could we rather use something like

```suggestion
func followCNAME(fqdn string, nameservers []string, fqdnChain ...string) (string, error) {
```

Reply:
I see that there is a new comment, it now makes more sense!

nit: still not interested in changing `updateDomainWithCName` with `followCNAME`? 😁  Update still feels a bit misleading 😓

```
// Follows the CNAME records and return the last non-CNAME fully qualified domain name that
// it finds. Returns an error when a CNAME loop is found in the CNAME chain.
func followCNAME(fqdn string, nameservers []string, fqdnChain ...string) (fqdn string, err error) {
}
```

(I used a named return argument `fqdn` to make sure the reader understands the "data flow": the user gives a fqdn and gets an fqdn back)

Reply:
Also, "there is a recursive CNAME" in the function comments is a bit confusing since a recusive CNAME might not be looping. Shall we rather say "there is a loop in the CNAME chain"?

Reply:
seems my changes got reveeted in some git mess